storage/libradb/src/lib.rs

```diff
@@ -192,20 +195,34 @@ impl LibraDB {
     /// `limit` events in the reverse order. Both cases are inclusive.
     fn get_events_by_event_access_path(
         &self,
-        access_path: &AccessPath,
+        query_path: &AccessPath,
         start_seq_num: u64,
         ascending: bool,
         limit: u64,
         ledger_version: Version,
-    ) -> Result<(Vec<EventWithProof>, Option<AccountStateWithProof>)> {
+    ) -> Result<(Vec<EventWithProof>, AccountStateWithProof)> {
         error_if_too_many_requested(limit, MAX_LIMIT)?;
 
         let get_latest = !ascending && start_seq_num == u64::max_value();
+        let account_state =
+            self.get_account_state_with_proof(query_path.address, ledger_version, ledger_version)?;
+        let account_resource = if let Some(account_blob) = &account_state.blob {
+            AccountResource::make_from(&(&account_blob.try_into()?))?
+        } else {
+            bail!("Nothing stored under address: {}", query_path.address);
+        };
+        let event_handle = if query_path.path == account_sent_event_path() {
+            account_resource.sent_events().as_access_path()?
```

Comment:
use `account_resource.get_event_handle_by_query_path()`, instead of this if-else.